gnucash/gnome-utils/gnc-autoclear.cpp

```diff
@@ -58,22 +59,22 @@ autoclear_quark (void)
     return g_quark_from_static_string ("autoclear");
 }
 
-typedef struct
+struct WorkItem
 {
-    GArray *workarray;
+    gint64 reachable_amount;
+    GList *list_of_splits;
+};
+
+struct sack_data
+{
+    std::vector<WorkItem*> workvector;
```

Comment:
Your use of std::vector is probably suboptimal because you store  pointers of `WorkItem`. That means the cpu can't benefit from cache locality. Additionally the list of splits is also a list of  pointers reducing the efficiency even more. It's actually impressive std::vector is already that efficient compared to a plain GHash in the current implementation.

Secondly I don't know if struct WorkItem is a movable type by default as no constructors, destructors and assignment operators are defined. In that case each potential move operation is performed as a copy operation instead.

All of that just to champion for std::vector rather than GArray*. As we move to C++ it's worth using the standard containers proposed and in almost all cases the std::vector is the fastest option as long as the cpu can benefit from cache locality.

Reply:
You're looking at an earlier commit whereby GArray was replaced with std::vector<Split*> so that it compiles. Would the latest commits improve things?